### PR TITLE
Update poi version to 5.2.3

### PR DESCRIPTION
**Pull request by josecsotomorales**

### CODE REVIEW

**Comment by dazfuller** [comment]
It looks like 5.2.3 brings about some log4j changes which are conflicting with older versions of Spark

**Comment by josecsotomorales** [comment]
@dazfuller not sure why it failed, it's specific to the build 3.3.1 only?

**Comment by dazfuller** [comment]
It seems to be stopping all of them when one of them fails, which isn't how it's meant to work. I'm going to run them 1 at a time in a while to see if it's just that one, or if it's pre 3.4

**Comment by dazfuller** [comment]
Okay, seems like it needs to pull in a later version of log4j to get it to work. I'm running some tests now but seems to be working by adding in these to the dependencies

```
  "org.apache.logging.log4j" % "log4j-core" % log4JVersion.value % Compile,
  "org.apache.logging.log4j" % "log4j-api" % log4JVersion.value % Compile
```

Need to test in on a Databricks instance or something though to make sure the shading rules don't mess up

File: build.sbt [hunk 1]
```diff
@@ -130,6 +130,6 @@ val commonSettings = Seq(
     }
   },
   scalaTestVersion := "3.2.16",
-  poiVersion := "5.2.2",
+  poiVersion := "5.2.3",
```
Comment (dazfuller) [inline build.sbt:133]: Look at doing the following, that should bring in the changes needed for pre-Spark 3.4:

Line 26, add the following: `val log4JVersion = settingKey[String]("Apache Log4J version")`

Line 65, add the following to the collection

```
  "org.apache.logging.log4j" % "log4j-core" % log4JVersion.value % Compile,
  "org.apache.logging.log4j" % "log4j-api" % log4JVersion.value % Compile
```

Line 133: Add this after `poiVersion`: `log4JVersion := "2.20.0",`

**Review by dazfuller — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 1 above)
Comment (josecsotomorales) [reply build.sbt:136]: Excellent, with that update we bring log4j as a project dependency so we don't get conflicts across spark versions, pushed the change request.

**Review by dazfuller — APPROVED** [review]
Not sure why the action didn't trigger on this. Guess that's another item in the to-do list

**Comment by josecsotomorales** [comment]
> Not sure why the action didn't trigger on this. Guess that's another item in the to-do list

Probably because of the target branch switched from `main` to `elastacloud:0_1_12-release`?

**Comment by dazfuller** [comment]
possibly, the action should kick in for opened and re-opened PRs though. Could be because I tweaked that mid-PR though

**Pull request merged**